com.unity.netcode.gameobjects/Runtime/Messaging/Messages/SnapshotDataMessage.cs

```diff
@@ -1,166 +1,55 @@
-using System;
 using Unity.Collections;
-using Unity.Collections.LowLevel.Unsafe;
 using UnityEngine;
 
 namespace Unity.Netcode
 {
     internal struct SnapshotDataMessage : INetworkMessage
     {
-        public int CurrentTick;
-        public ushort Sequence;
+        internal FastBufferWriter WriteBuffer;
+        internal FastBufferReader ReadBuffer;
+        private int m_BufferSize;
 
-        public ushort Range;
-
-        public byte[] SendMainBuffer;
-        public NativeArray<byte> ReceiveMainBuffer;
-
-        public struct AckData
+        // a constructor with an unused parameter is used because C# doesn't allow parameter-less constructors
+        public SnapshotDataMessage(int unused)
         {
-            public ushort LastReceivedSequence;
-            public ushort ReceivedSequenceMask;
+            m_BufferSize = 10000;
+            WriteBuffer = new FastBufferWriter(m_BufferSize, Allocator.Temp);
+            ReadBuffer = new FastBufferReader(WriteBuffer, Allocator.Temp);
```

Comment:
I'd also suggest initializing these only when they're used. Creating the WriteBuffer when the struct is created for send, while leaving ReadBuffer uninitialized, and likewise creating ReadBuffer within Deserialize().

Some things to be aware of:

FastBufferReader and FastBufferWriter both need to be disposed. This is one reason for creating the buffer outside of the struct, so you can `using` it and dispose it after the message is sent; alternatively, it can be disposed at the end of `Serialize()` but that risks a leak if any exceptions happen between creation and the end of Serialize().

On the receiving end, messages are created with an empty constructor, so this won't actually be called for receiving the message at all. So this is being created and unused when sending, and not created when receiving. (But it doesn't need to be created in either case since it's set to the passed in `reader` parameter when receiving.)